zio-telemetry-example-proxy/src/main/scala/zio/telemetry/example/proxy/Server.scala

```diff
@@ -0,0 +1,72 @@
+package zio.telemetry.example.proxy
+
+import cats.effect.ExitCode
+import io.circe.generic.auto._
+import io.circe.syntax._
+import io.circe.{ Decoder, Encoder }
+import io.opentracing.Tracer
+import io.opentracing.propagation.Format.Builtin.{ HTTP_HEADERS => HttpHeadersFormat }
+import io.opentracing.propagation._
+import org.http4s._
+import org.http4s.circe._
+import org.http4s.client.blaze.BlazeClientBuilder
+import org.http4s.dsl.io._
+import org.http4s.server.Router
+import org.http4s.server.blaze.BlazeServerBuilder
+import org.http4s.syntax.kleisli._
+import zio._
+import zio.clock.Clock
+import zio.interop.catz._
+import zio.telemetry.example.proxy.config.{ Configuration, ServiceConfig }
+import zio.telemetry.example.proxy.http.StatusResponse
+import zio.telemetry.opentracing._
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+import scala.concurrent.ExecutionContext
+
+object Server extends CatsApp {
+
+  type AppTask[A] = ZIO[Clock, Throwable, A]
+
+  def app(config: ServiceConfig, tracer: Tracer): HttpApp[AppTask] = {
+    implicit def decoder[A: Decoder]: EntityDecoder[AppTask, A] = jsonOf
+
+    implicit def encoder[A: Encoder]: EntityEncoder[AppTask, A] = jsonEncoderOf
+
+    val routes = HttpRoutes.of[AppTask] {
+      case GET -> Root / "statuses" =>
+        JaegerTracing.makeService(tracer).use { env =>
+          (for {
+            _ <- env.telemetry.root("/statuses")
+            _ <- env.telemetry.tag("proxy-tag", "proxy-tag-value")
+            _ = env.telemetry.inject(HttpHeadersFormat, new TextMapAdapter(mutable.Map.empty[String, String].asJava))
```

Comment:
What is the motivation for inject here? The textMap is immediately discarded. Btw, It might be niche to add convenience methods e.g. for HttpHeadersFormat, which do not side effecting update a text map, but return a UIO[Map[String, String]].